### PR TITLE
Upgrade Rails to version 7.2.1

### PR DESCRIPTION
**Pull request by unoduetre**
⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

Follow [these steps](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html) if you are doing a Rails upgrade.

## What

Upgrade government-frontend to use the latest version of Rails (7.2.1).

## Why

[Trello card](https://trello.com/c/EfS9DHDw/2784-update-our-applications-to-rails-72-l)

### CODE REVIEW

File: bin/rubocop [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env ruby
```
Comment (unoduetre) [inline bin/rubocop:1]: Added in order to support the config option.

File: config/initializers/permissions_policy.rb [hunk 2]
```diff
@@ -1,11 +1,13 @@
+# Be sure to restart your server when you modify this file.
```
Comment (unoduetre) [inline config/initializers/permissions_policy.rb:1]: Changes suggested by rails app:update.

File: public/404.html [hunk 3]
```diff
@@ -4,21 +4,21 @@
   <title>The page you were looking for doesn't exist (404)</title>
```
Comment (unoduetre) [inline public/404.html:4]: Changes suggested by rails app:update.

File: public/500.html [hunk 4]
```diff
@@ -4,21 +4,21 @@
   <title>We're sorry, but something went wrong (500)</title>
```
Comment (unoduetre) [inline public/500.html:4]: Changes suggested by rails app:update.

**Comment by unoduetre** [comment]
The default 7.2 config enables [YJIT](https://guides.rubyonrails.org/configuring.html#config-yjit) if Ruby version is at least 3.3.

**Comment by unoduetre** [comment]
I have removed the icon files.

**Comment by unoduetre** [comment]
I'm updating the ticket to upgrade to 7.2.1

**Comment by unoduetre** [comment]
> I'm updating the ticket to upgrade to 7.2.1

Done

File: config/environments/development.rb [hunk 5]
```diff
@@ -52,10 +52,13 @@
...
 
   config.hosts.clear
 
-  # Uncomment if you wish to allow Action Cable access from any origin.
-  # config.action_cable.disable_request_forgery_protection = true
+  # Raise error when a before_action's only/except options reference missing actions.
+  config.action_controller.raise_on_missing_callback_actions = false
+
+  # Apply autocorrection by RuboCop to files generated by `bin/rails generate`.
+  config.generators.apply_rubocop_autocorrect_after_generate!
```
Comment (unoduetre) [inline config/environments/development.rb:63]: Enabled so our rubocop rules apply automatically.

File: config/environments/development.rb [hunk 6]
```diff
@@ -52,10 +52,13 @@
   # config.i18n.raise_on_missing_translations = true
 
   # Annotate rendered view with file names.
-  # config.action_view.annotate_rendered_view_with_filenames = true
+  config.action_view.annotate_rendered_view_with_filenames = true
```
Comment (beccapearce) [inline config/environments/development.rb:55]: Not a blocker, but would be nice to call out why we're enabling this

**Review by beccapearce — APPROVED** [review]
Looks good! 
One non-blocking comment :)

File: config/environments/development.rb (hunk 6 above)
Comment (unoduetre) [reply config/environments/development.rb:55]: It is a change suggested by app:update.

As it only applies to the development environment, it's safe to apply with no impact on tests or production.

It helps in debugging. Here is the description of what it does: https://www.bigbinary.com/blog/rails-6-1-adds-annotate_rendered_view_with_filenames-to-annotate-html-output

What is your opinion on that?

Comment (beccapearce) [reply config/environments/development.rb:55]: I think 'It helps in debugging.' is enough. Just so people looking back can understand why we decided to uncomment it.

Comment (beccapearce) [reply config/environments/development.rb:55]: Also I think this conversation is probably enough documentation too! So I think all good to be merged! 💪🏻

**Pull request merged**